homeassistant/components/switcher_kis/__init__.py

```diff
@@ -90,4 +102,28 @@ def device_updates(timestamp: Optional[datetime]) -> None:
 
     async_track_time_interval(hass, device_updates, timedelta(seconds=4))
 
+    async def async_set_auto_off_service(service: ServiceCallType) -> None:
+        """Use for handling setting device auto-off service calls."""
+        from aioswitcher.api import SwitcherV2Api
+        async with SwitcherV2Api(hass.loop, device_data.ip_addr, phone_id,
```

Comment:
You should validate that the user has at least access to control a single entity of this integration.

Reply:
Thank you very much, I'll get right on it!
:-)

Reply:
Resolved with the use of `async_listen_platform` in [this commit](https://github.com/home-assistant/home-assistant/pull/23477/commits/2a5a002d7452d70bbfb32785e7cfa3d7c1016848).

Reply:
@MartinHjelmare 
Oh yes I see that now.
I'll have a closer look tonight and push the adjustments.
Thanks!

Reply:
Resolved in this [commit](https://github.com/home-assistant/home-assistant/pull/23477/commits/1addd0160afd6c95c5c58fb36cd6ebaf9ad88660).

Reply:
Although I must say, I'm not that comfortable with the way I incorporated the access rights verification.

I think we should maybe add the ability to set permission policies for users at the integration level.
I this case, both services `switcher_kis.set_auto_off` and `switcher_kis.update_device_name` perform actions on the integration's behalf, and not quite on the `switch`'s behalf considering the `switch` itself actually performs turn on/off actions on the integration's behalf.

(Please keep in mind there's a `sensor` platform in this integration's near future, that's why I'm referencing the `switch` platform as just a part of the integration.)

None of these two services actually requires the `switch` entity or the future `sensor` entities, I actually added the `entity_id` parameter to the services schema just for the permission check.
Which is ok to do, but it's not really accurate seeing that I might want a user to have permission to turn on or off the `switch`, but not change the device's name at the integration level.

Reply:
Now that I think about it...
I should probably add a component name verification before the permission verification.
Otherwise a user can pass any permitted entity and he will pass the verification process.

Reply:
Added `domain` and `entity_id` validation in [this](https://github.com/home-assistant/home-assistant/pull/23477/commits/0a1e62c3a1473737baf6b20025992955e1742c22) commit.
The `domain` validation might seem redundant, but it won't be once I'll integrate the `sensor` platform representing the device's schedules.
For now, I can easily remove it if necessary.

Reply:
@MartinHjelmare 
Any thoughts?
:-)